scripts/include/hook-common

```diff
@@ -0,0 +1,42 @@
+default_layouts="us"
+layouts_file="$SNAP/etc/layouts.txt"
+osk_layout_file="$SNAP_DATA/osk-layout.txt"
+
+# Use the first argument to join all the remaining ones
+function join {
+  local IFS="$1"
+  shift
+  echo "$*"
+}
+
+# Loop through list of layouts ensuring entries are valid and comma separated
+# Prints all the valid layouts and returns an error if any was not.
+function validate_layouts {
+  local error=0
+  for layout in $(snapctl get layout | tr ',' ' '); do
+    if grep -Fxq "$layout" "$layouts_file"; then
+      echo ${layout}
+    else
+      echo "Error: $layout is not a valid layout ID." >&2
+      echo "       See $SNAP_NAME.list-layouts for a list of valid layouts." >&2
+      error=1
+    fi
+  done
+  return $error
+}
+
+# Prints "yes" or "no" depending on whether the file changed or not
+function set_layouts {
```

Comment:
The arguments to this function should be documented, and it might be better if it takes a comma-list instead of multiple arguments.